### DISCUSSION TITLE
Signac + Cicero

### DISCUSSION THREAD

**Discussion by mspivakov** [stuart-lab/signac#840] [Ideas]
Just to say that I've found the route of integrating Signac with Cicero proposed in the docs impossible to implement on my system, because Monocle3's dependencies proved to be a massive pain to install. It took me a while to realise I could use the Bioconductor versions of Monocle and Cicero instead by running:

```
library(Seurat)
library(Signac)
library(cicero) # the Bioconductor version
# library(SeuratWrappers) not needed
 
cds = as.CellDataSet(x=seuratATACObject) # as opposed to as.cell_data_set 
cds.cicero=  make_cicero_cds(cds, reduced_coordinates = Reductions(seuratATACObject, "umap.atac")@cell.embeddings)
genome = seqlengths(BSgenome.Hsapiens.UCSC.hg38) 
genome.df <- data.frame("chr" = names(genome), "length" = genome)
conns <- run_cicero(cds.cicero, genomic_coords = genome.df, sample_num = 100)
 
```

Am I right that this is fine (although potentially misses out on new functionality of Cicero/Monocle from the github versions)? 
If so, it might be worth presenting this route as an alternative in the docs to save time for people like me who have trouble installing Monocle3 and outline whatever caveats this entails. 

Thanks!

**Comment by cristanchoa** [comment]
I am having the same issue, but couldn't get it work with what I thought was the Monocle/cicero combination. Can you please list what you installed? Thank you! This has been driving me crazy.

**Reply by mspivakov** [reply]
Totally sympathise, I spent weeks (or and off) trying to fix this. I have just run BiocManager::install("monocle") and BiocManager::install("cicero") after having installed Seurat and Signac.

**Comment by timoast** [comment]
This uses Monocle2 rather than Monocle3, I'm not sure what differences that may create in the Cicero analysis. You can ask the Cicero/Monocle developers instead whether they recommend using Cicero+Monocle2, but from what I can tell from reading their [docs](https://cole-trapnell-lab.github.io/cicero-release/docs/), it should be fine.
